SplitBrainHandlerTest.testClusterMerge_ignoresLiteMembers

```
java.lang.AssertionError: Cluster size is not correct. Expected: 2 Actual: 5
	at org.junit.Assert.fail(Assert.java:88)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSize(HazelcastTestSupport.java:870)
	at com.hazelcast.cluster.SplitBrainHandlerTest.testClusterMerge_ignoresLiteMembers(SplitBrainHandlerTest.java:521)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-nightly/com.hazelcast$hazelcast/1355/testReport/junit/com.hazelcast.cluster/SplitBrainHandlerTest/testClusterMerge_ignoresLiteMembers/

*master*

Reopening as it failed in the nighly build: https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-nightly-lab/43/testReport/junit/com.hazelcast.cluster/SplitBrainHandlerTest/testClusterMerge_ignoresLiteMembers/

Stacktrace:
```
com.hazelcast.spi.exception.WrongTargetException: WrongTarget! this: [127.0.0.1]:5705, target: null, partitionId: 31, replicaIndex: 0, operation: com.hazelcast.map.impl.operation.PutOperation, service: hz:impl:mapService
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.newTargetNullException(Invocation.java:631)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.initInvocationTarget(Invocation.java:243)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:540)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.access$300(Invocation.java:97)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation$InvocationRetryTask.run(Invocation.java:753)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
	at ------ submitted from ------.(Unknown Source)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolve(InvocationFuture.java:127)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrowIfException(InvocationFuture.java:79)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:155)
	at com.hazelcast.map.impl.proxy.MapProxySupport.invokeOperation(MapProxySupport.java:425)
	at com.hazelcast.map.impl.proxy.MapProxySupport.putInternal(MapProxySupport.java:390)
	at com.hazelcast.map.impl.proxy.MapProxyImpl.put(MapProxyImpl.java:131)
	at com.hazelcast.map.impl.proxy.MapProxyImpl.put(MapProxyImpl.java:122)
	at com.hazelcast.cluster.SplitBrainHandlerTest.testClusterMerge_ignoresLiteMembers(SplitBrainHandlerTest.java:540)
```

Standard output:
```
00:01:41,701  WARN |testClusterMerge_ignoresLiteMembers| - [MulticastJoiner] hz.lite1.cached.thread-2 - [127.0.0.1]:5701 [cfevbhxzza] [3.10-SNAPSHOT] Error during join check!
java.util.concurrent.ExecutionException: com.hazelcast.spi.exception.RetryableIOException: Packet not sent to -> [127.0.0.1]:5704
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrowIfException(InvocationFuture.java:92) ~[classes/:?]
	at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:171) ~[classes/:?]
	at com.hazelcast.internal.cluster.impl.AbstractJoiner.sendSplitBrainJoinMessage(AbstractJoiner.java:262) [classes/:?]
	at com.hazelcast.internal.cluster.impl.AbstractJoiner.sendSplitBrainJoinMessageAndCheckResponse(AbstractJoiner.java:227) [classes/:?]
	at com.hazelcast.internal.cluster.impl.MulticastJoiner.searchForOtherClusters(MulticastJoiner.java:128) [classes/:?]
	at com.hazelcast.internal.cluster.impl.SplitBrainHandler.searchForOtherClusters(SplitBrainHandler.java:75) [classes/:?]
	at com.hazelcast.internal.cluster.impl.SplitBrainHandler.run(SplitBrainHandler.java:42) [classes/:?]
	at com.hazelcast.spi.impl.executionservice.impl.DelegateAndSkipOnConcurrentExecutionDecorator$DelegateDecorator.run(DelegateAndSkipOnConcurrentExecutionDecorator.java:66) [classes/:?]
	at com.hazelcast.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:227) [classes/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_161]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_161]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_161]
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64) [classes/:?]
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80) [classes/:?]
Caused by: com.hazelcast.spi.exception.RetryableIOException: Packet not sent to -> [127.0.0.1]:5704
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeRemote(Invocation.java:580) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:557) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:518) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:208) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:60) ~[classes/:?]
	at com.hazelcast.internal.cluster.impl.AbstractJoiner.sendSplitBrainJoinMessage(AbstractJoiner.java:260) ~[classes/:?]
	at com.hazelcast.internal.cluster.impl.AbstractJoiner.sendSplitBrainJoinMessageAndCheckResponse(AbstractJoiner.java:227) ~[classes/:?]
	at com.hazelcast.internal.cluster.impl.MulticastJoiner.searchForOtherClusters(MulticastJoiner.java:128) ~[classes/:?]
	at com.hazelcast.internal.cluster.impl.SplitBrainHandler.searchForOtherClusters(SplitBrainHandler.java:75) ~[classes/:?]
	at com.hazelcast.internal.cluster.impl.SplitBrainHandler.run(SplitBrainHandler.java:42) ~[classes/:?]
	at com.hazelcast.spi.impl.executionservice.impl.DelegateAndSkipOnConcurrentExecutionDecorator$DelegateDecorator.run(DelegateAndSkipOnConcurrentExecutionDecorator.java:66) ~[classes/:?]
	at com.hazelcast.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:227) ~[classes/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_161]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_161]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_161]
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64) ~[classes/:?]
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80) ~[classes/:?]
	at ------ submitted from ------.(Unknown Source) ~[?:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolve(InvocationFuture.java:127) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrowIfException(InvocationFuture.java:79) ~[classes/:?]
	... 13 more
00:01:41,701  WARN |testClusterMerge_ignoresLiteMembers| - [MulticastJoiner] hz.data2.cached.thread-2 - [127.0.0.1]:5704 [cfevbhxzza] [3.10-SNAPSHOT] Error during join check!
java.util.concurrent.ExecutionException: com.hazelcast.spi.exception.RetryableIOException: Packet not sent to -> [127.0.0.1]:5701
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrowIfException(InvocationFuture.java:92) ~[classes/:?]
	at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:171) ~[classes/:?]
	at com.hazelcast.internal.cluster.impl.AbstractJoiner.sendSplitBrainJoinMessage(AbstractJoiner.java:262) [classes/:?]
	at com.hazelcast.internal.cluster.impl.AbstractJoiner.sendSplitBrainJoinMessageAndCheckResponse(AbstractJoiner.java:227) [classes/:?]
	at com.hazelcast.internal.cluster.impl.MulticastJoiner.searchForOtherClusters(MulticastJoiner.java:128) [classes/:?]
	at com.hazelcast.internal.cluster.impl.SplitBrainHandler.searchForOtherClusters(SplitBrainHandler.java:75) [classes/:?]
	at com.hazelcast.internal.cluster.impl.SplitBrainHandler.run(SplitBrainHandler.java:42) [classes/:?]
	at com.hazelcast.spi.impl.executionservice.impl.DelegateAndSkipOnConcurrentExecutionDecorator$DelegateDecorator.run(DelegateAndSkipOnConcurrentExecutionDecorator.java:66) [classes/:?]
	at com.hazelcast.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:227) [classes/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_161]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_161]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_161]
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64) [classes/:?]
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80) [classes/:?]
Caused by: com.hazelcast.spi.exception.RetryableIOException: Packet not sent to -> [127.0.0.1]:5701
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeRemote(Invocation.java:580) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:557) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:518) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:208) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:60) ~[classes/:?]
	at com.hazelcast.internal.cluster.impl.AbstractJoiner.sendSplitBrainJoinMessage(AbstractJoiner.java:260) ~[classes/:?]
	at com.hazelcast.internal.cluster.impl.AbstractJoiner.sendSplitBrainJoinMessageAndCheckResponse(AbstractJoiner.java:227) ~[classes/:?]
	at com.hazelcast.internal.cluster.impl.MulticastJoiner.searchForOtherClusters(MulticastJoiner.java:128) ~[classes/:?]
	at com.hazelcast.internal.cluster.impl.SplitBrainHandler.searchForOtherClusters(SplitBrainHandler.java:75) ~[classes/:?]
	at com.hazelcast.internal.cluster.impl.SplitBrainHandler.run(SplitBrainHandler.java:42) ~[classes/:?]
	at com.hazelcast.spi.impl.executionservice.impl.DelegateAndSkipOnConcurrentExecutionDecorator$DelegateDecorator.run(DelegateAndSkipOnConcurrentExecutionDecorator.java:66) ~[classes/:?]
	at com.hazelcast.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:227) ~[classes/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_161]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_161]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_161]
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64) ~[classes/:?]
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80) ~[classes/:?]
	at ------ submitted from ------.(Unknown Source) ~[?:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolve(InvocationFuture.java:127) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrowIfException(InvocationFuture.java:79) ~[classes/:?]
	... 13 more
```

--- a/hazelcast/src/test/java/com/hazelcast/cluster/SplitBrainHandlerTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/cluster/SplitBrainHandlerTest.java
@@ -536,6 +536,9 @@ public class SplitBrainHandlerTest extends HazelcastTestSupport {
         assertClusterSizeEventually(3, lite1, lite2, data1);
         assertClusterSizeEventually(2, data2, data3);
 
+        waitAllForSafeState(lite1, lite2, data1);
+        waitAllForSafeState(data2, data3);
+
         data1.getMap("default").put(1, "cluster1");
         data3.getMap("default").put(1, "cluster2");
 
@@ -547,9 +550,9 @@ public class SplitBrainHandlerTest extends HazelcastTestSupport {
         unblockCommunicationBetween(lite2, data3);
         unblockCommunicationBetween(data1, data3);
 
-        assertOpenEventually(mergeLatch, 120);
+        assertOpenEventually(mergeLatch);
         assertClusterSizeEventually(5, lite1, lite2, data1, data2, data3);
-
+        waitAllForSafeState(lite1, lite2, data1, data2, data3);
         assertEquals("cluster1", lite1.getMap("default").get(1));
     }
 

